kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/predicates/In.java

```diff
@@ -116,8 +116,11 @@ public String toString() {
   }
 
   public String toString(Schema schema) {
-    return Joiner.on(',').join(
-        Iterables.transform(set, new ToString<T>(schema)));
+    // we want the string this returns to be consistent
+    // ensure the set is sorted before calling toString on it.
+
+    Iterable<String> strings = Iterables.transform(set, new ToString<T>(schema));
+    return Joiner.on(',').join(Sets.newTreeSet(strings));
```

Comment:
Thanks for opening an issue for it. I think the caller should request normalization, so that would be great.